x-pack/plugins/fleet/server/services/agents/actions.ts

```diff
@@ -302,6 +319,29 @@ export async function cancelAgentAction(esClient: ElasticsearchClient, actionId:
         .getLogger()
         .info(`Errors while bulk updating agents for cancel action: ${JSON.stringify(errors)}`);
     }
+  };
+
+  for (const action of upgradeActions) {
+    await updateAgentsToHealthy(action);
+  }
+
+  // find missing batch
```

Comment:
at the end of cancel, doing one more query on upgrade action to find those docs that were created by a concurrent upgrade action
this solves the issue of cancelling all batches
tested locally with 980 agents, 100 batch size, cancelling shortly after upgrade. Agents go back to Healthy except for 20, which are immediately upgraded

<img width="2840" alt="image" src="https://user-images.githubusercontent.com/90178898/216377678-28941bcf-04c5-4dd0-8e83-daf5e5d6de45.png">

Reply:
nit: it may be helpful to expand the code comment to include more of this useful information ^